kpops/component_handlers/kafka_connect/connect_wrapper.py

```diff
@@ -121,16 +121,26 @@ def update_connector_config(
             self.update_connector_config(connector_name, kafka_connect_config)
         raise KafkaConnectError(response)
 
+    def get_connector_config(self, connector_name: str, config: KafkaConnectConfig):
+        connector_config = config.dict(exclude_none=True, exclude_unset=True)
+        if connector_config.get("name") and connector_name != connector_config.get(
+            "name"
+        ):
+            raise ValueError("App config name should be the same as component name")
+        connector_config["name"] = connector_name
+        return connector_config
+
     def validate_connector_config(
-        self, kafka_connect_config: KafkaConnectConfig
+        self, connector_name: str, kafka_connect_config: KafkaConnectConfig
     ) -> list[str]:
         """
         Validate connector config using the given configuration
         :param connector_name: Name of the created connector
         :param kafka_connect_config: Configuration parameters for the connector.
         :return:
         """
-        config_json = kafka_connect_config.dict(exclude_none=True, exclude_unset=True)
+
+        config_json = self.get_connector_config(connector_name, kafka_connect_config)
         connector_class = ConnectWrapper.get_connector_class_name(config_json)
 
         response = requests.put(
```

Comment:
Please add  a test that verifies the correct payload for this endpoint

Reply:
isn't this already what you mean ? https://github.com/bakdata/kpops/blob/fix/kafka_name_match/tests/component_handlers/kafka_connect/test_connect_wrapper.py#L448

Reply:
yeah but you should remove the name in order to see that it gets added